### ISSUE TITLE
Add hyperparameter sweep command (grid + random search)

### ISSUE THREAD

**Issue by mmgalushka** [mmgalushka/cvbench#71]
## What

Add a `cvbench sweep` command that runs many training trials over a range of hyperparameter values and reports the best one, so tuning no longer means hand-editing `cvbench train` flags and re-running repeatedly.

## Why

Finding a good config today means manually running `cvbench train` with different `--lr`, `--backbone`, `--batch-size`, `--optimizer`, `--dropout`, etc., then eyeballing `cvbench runs list` / `cvbench runs best`. The `experiments/` directory already shows this pattern (many `effnet_b0_lr1e3_*`, `det_resnet_18_lr1e4_*` runs). A sweep automates the loop and the selection.

### CLI surface — `sweep` mirrors `train`

`sweep` takes the **same flags as `train`**, each accepting a list. "A sweep is `train` with comma-separated values" — same flag names, same `type:params` micro-syntax already parsed in `src/cvbench/cli/train.py` (`_parse_optimizer`, `_parse_loss`, `_parse_lr_scheduler` are just mapped over a split list).

```bash
cvbench sweep data/shapes \
  --strategy grid \
  --lr 1e-3,1e-4,1e-5 \
  --backbone efficientnet_b0,resnet_18 \
  --batch-size 16,32 \
  --optimizer adam,sgd:momentum=0.9 \
  --metric val_loss
```

- **`--strategy grid`** — Cartesian product of all list-valued flags.
- **`--strategy random --trials N`** — draw N configs; choice flags sampled uniformly, ranges written with `:` (`--dropout 0.1:0.5`) sampled over the interval. `:` ranges are rejected under `grid`.
- A flag given a single value is fixed across all trials (same as `train`).
- `--from <exp_dir>` supplies the baseline config; swept flags override per trial.

No YAML spec input — flags are the primitive.

### Presets — curated flag-sets

Hand-building a grid is laborious, so ship named presets as **package data** (`src/cvbench/presets/sweep/*.yaml`, project-maintained, not user-authored):

```bash
cvbench sweep data/shapes --preset default
cvbench sweep data/shapes --preset default --lr 1e-3,1e-4,1e-5   # widen one axis; explicit flags win
cvbench sweep data/shapes --preset default --show               # print resolved grid + trial count, don't run
cvbench sweep presets                                            # list available presets
```

- `quick` — smoke test, ~4 trials (2 LRs x 2 backbones)
- `default` — LR x backbone x optimizer, ~12-18 trials
- `thorough` — adds batch-size, dropout, augmentation on/off

Each preset resolves per task (see below). Presets bake in assumptions (epoch count, dataset scale) — kept small, with documented trial count and rough runtime; detection presets especially can run hours. Presets need occasional review so they don't drift from current good defaults.

### Works for both tasks

Task is auto-detected from the data directory (`detect_task_name`), exactly like `train` — no `--task` flag, no branching in the sweep loop. Two task-aware touches only:

1. **Default `--metric`** — classification defaults to `val_loss` (min) / accepts `val_accuracy`; detection defaults to mAP@50 (max), i.e. the existing `RunConfig.test_metric = "map50"`. Min-vs-max direction follows from the metric name. `best_experiment()` already takes a metric argument and is reused for final selection.
2. **Sweepable params = `train`'s surface for that task.** Shared knobs (`--lr`, `--backbone`, `--batch-size`, `--optimizer`, `--dropout`, `--augmentation`, `--epochs`) cover both. Detection-only knobs (`noobj_weight`, `conf_threshold`, `iou_threshold`, ...) aren't `train` CLI flags today, so they're out of scope for the sweep until they are.

### Execution

Sequential, one trial at a time (single-sandbox). **No resume** — an interrupted sweep is restarted from scratch. Each trial:

1. builds a `CVBenchConfig` via the existing `build_config()` (baseline `--from` + per-trial overrides),
2. runs `run_training()` into `experiments/<sweep_name>/trial_001/`, `trial_002/`, ...,
3. evaluates on the val split and records the selection metric.

### Output

- `experiments/<sweep_name>/sweep.yaml` — manifest listing every trial's params + result (a record, not an input format).
- End-of-sweep summary table + pointer to the best trial (via `best_experiment()`).
- `cvbench sweep status <sweep_dir>` — reprint the summary table and current best.

## Non-goals

- No Bayesian / guided optimization (grid + random only).
- No parallel trial execution.
- No sweep resume/checkpointing.
- No WebUI integration — surfacing sweep results in the WebUI is a follow-up issue.
- No new detection-specific tunable flags — separate issue if wanted.

## Acceptance criteria

- `cvbench sweep` runs a grid and a random sweep end to end for both a classification and a detection data layout, producing one experiment dir per trial under `experiments/<sweep_name>/`.
- `--preset` resolves to a task-appropriate grid; `--preset ... --show` prints it without training; explicit flags override preset axes.
- `sweep.yaml` manifest records each trial's params and selection metric.
- Best trial identified by the configured (or task-default) metric and reported at the end.
- `cvbench sweep status <sweep_dir>` reprints the summary.
- Invalid inputs fail fast with clear messages (`:` range under `grid`, unknown flag, `random` without `--trials`).
- `README.md` and `helper.sh` updated to document `sweep` and its presets.
- Tests cover list parsing, grid expansion, random sampling, preset resolution, and best-trial selection.

**Comment by mmgalushka** [comment]
Resolved by the merged PR for issue #71.
